### PR TITLE
[NP-3232] Fix bugs from NP-3188

### PR DESCRIPTION
**Pull request by KernelDeimos**
Introduces the following changes:
- Fix issues caused by NP-3188
- Add a stdlib function to make reduction promises more readable
- Rename `dataController` property to `wao`

### CODE REVIEW

**Review by jlhughes — APPROVED** [review]

File: src/foam/core/stdlib.js [hunk 1]
```diff
@@ -123,6 +123,17 @@ foam.LIB({
 });
 
 
+foam.LIB({
+  name: 'foam.Promise',
+  methods: [
+    function reduce(arry, fn, parentPromise) {
```
Comment (kgrgreer) [inline src/foam/core/stdlib.js]: A better name for parentPromise would be opt_parentPromise to indicate that it's optional.

Comment (kgrgreer) [inline src/foam/core/stdlib.js]: Add documentation.

File: src/foam/u2/crunch/FlatteningCapabilityWizardlet.js [hunk 2]
```diff
@@ -56,6 +56,18 @@ foam.CLASS({
 
       // Return true to prevent wizardlet from being pushed normally
       return true;
+    },
+    async function save() {
+      await foam.Promise.reduce(this.delegates, d => d.save());
```
Comment (kgrgreer) [inline src/foam/u2/crunch/FlatteningCapabilityWizardlet.js]: Do you really want a 'reduce' operation? Do you really compare to combine the results? Or, would a forEach/forAll/all method be more understandable?

Comment (KernelDeimos) [reply src/foam/u2/crunch/FlatteningCapabilityWizardlet.js]: Yes, a reduce is desired. I've been using the following technique whenever I expect promises to execute after each previous promise is complete:
```
newPromise = objects.reduce((p, object) =>
  p.then(() => object.someAsyncMethod()), Promise.resolve());
```

I could use `Promise.all`, but then wizardlets are saved an an arbitrary order which we don't want here - we actually want these asynchronous behaviours to occur synchronously.

The reason for adding this to `stdlib` is that I find I need this a lot, but it's not very readable when I write it as I've done in the example above.

Comment (kgrgreer) [reply src/foam/u2/crunch/FlatteningCapabilityWizardlet.js]: Reduce doesn't imply that they are executed in sequence, so maybe something like inSeq() or inSequence() or aSeq() (the same as FOAM1)?

Comment (KernelDeimos) [reply src/foam/u2/crunch/FlatteningCapabilityWizardlet.js]: If I read `aSeq` I would believe it to mean "not sequential" but I like `inSeq`

Comment (KernelDeimos) [reply src/foam/u2/crunch/FlatteningCapabilityWizardlet.js]: Oh, how about just `inOrder`?

**Pull request merged**